### PR TITLE
Set inital value in useLocaleStorageHook

### PR DESCRIPTION
**Pull request by Doaa-Ismael**
call fetcher in `useLocaleStorage` hook if there is no default value provided

### CODE REVIEW

File: src/hooks/useLocalStorage.ts [hunk 1]
```diff
@@ -20,15 +20,19 @@ const useLocalStorage = <T>(key: string, defaultValue: T|null = null): LocalStor
     // If value not found from store, try to fetch it from local storage
     if (!isServerSide()) {
       const localStorageValue = window.localStorage.getItem(persistentValueKey);
-      if (localStorageValue != null) { value = JSON.parse(localStorageValue); }
+      if (localStorageValue !== null && localStorageValue !== undefined && localStorageValue !== 'undefined') {
```
Comment (abdelhamid-attaby) [inline src/hooks/useLocalStorage.ts:23]: `!= null` is equivalent to `!== null & !== undefined`

Comment (abdelhamid-attaby) [reply src/hooks/useLocalStorage.ts:23]: image

**Review by AmrSaber — COMMENTED** [review]
What you did will solve the problem, but I think there is no need for the fetcher at all for this hook (just like the global state hook) we fetch the value from the local storage for the first time, then we basically manage a state + save value to local storage when that state changes.

So I think we could remove the whole `persistentFetcher` and replace lines 7-30 with...
```js
const initialValue = isServerSide() ? defaultValue : window.localStorage.getItem(persistentValueKey) ?? defaultValue;
```

Then we can pass the fetcher as `null` to `userSWR` on line 33. This will work exactly as it does in the global state hook. It will also make the hook simpler.

**Comment by Doaa-Ismael** [comment]
@AmrSaber Okay it makes sense, I will remove the fetcher, but I think `defaultValue` should have precedence more than locale storage right?

**Comment by AmrSaber** [comment]
@Doaa-Ismael I think it makes more sense to have the stored value first, as when the user opens the site, they expect to see what they have set from the previous session and not the set default value.

**Review by AmrSaber — DISMISSED** [review]

**Comment by AmrSaber** [comment]
Please update package version in package-lock.json too

**Review by AmrSaber — APPROVED** [review]

**Pull request merged**